core/autoscaling.go

```diff
@@ -20,16 +27,149 @@ type autoScalingGroup struct {
 
 	// spot instance requests generated for the current group
 	spotInstanceRequests []*spotInstanceRequest
+	minOnDemand          int64
 }
 
-func (a *autoScalingGroup) process() {
+func (a *autoScalingGroup) loadPercentageOnDemand(tagValue *string) (int64, bool) {
+	percentage, err := strconv.ParseFloat(*tagValue, 64)
+	if err != nil {
+		logger.Printf("Error with ParseFloat: %s\n", err.Error())
+	} else if percentage == 0 {
+		logger.Printf("Loaded MinOnDemand value to %f from tag %s\n", percentage, OnDemandPercentageLong)
+		return int64(percentage), true
+	} else if percentage > 0 && percentage <= 100 {
+		instanceNumber := float64(len(a.instances.catalog))
+		onDemand := int64(math.Floor((instanceNumber * percentage / 100.0) + .5))
```

Comment:
I see this works out in the tests, but why add 0.5 here?

Shouldn't `math.Floor()` round up to the correct value without it?